modules/checkout/src/Plugin/Commerce/CheckoutPane/OrderSummary.php

```diff
@@ -70,21 +89,36 @@ public function submitConfigurationForm(array &$form, FormStateInterface $form_s
 
     if (!$form_state->getErrors()) {
       $values = $form_state->getValue($form['#parents']);
-      $this->configuration['view'] = $values['view'];
+      if (!$values['use_views']) {
+        $this->configuration['view'] = NULL;
+      }
+      else {
+        $this->configuration['view'] = $values['view'];
+      }
     }
   }
 
   /**
    * {@inheritdoc}
    */
   public function buildPaneForm(array $pane_form, FormStateInterface $form_state, array &$complete_form) {
-    $pane_form['view'] = [
-      '#type' => 'view',
-      '#name' => $this->configuration['view'],
-      '#display_id' => 'default',
-      '#arguments' => [$this->order->id()],
-      '#embed' => TRUE,
-    ];
+    if (!$this->configuration['view']) {
```

Comment:
Stylistic nitpick: Feels weird for the first condition to be a negation, it requires more energy to mentally parse. I'd always start with if ($this->configuration['view']) then handle the other case in the else.